extensions/integration-tests/src/schemaCompare.test.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import 'mocha';
+import * as azdata from 'azdata';
+import * as utils from './utils';
+const path = require('path');
+import { context } from './testContext';
+import assert = require('assert');
+import { getStandaloneServer } from './testConfig';
+import { exists, readFileSync, unlinkSync } from 'fs';
+
+if (context.RunTest) {
+	suite('Schema compare integration test suite', () => {
+		suiteSetup(async function () {
+			await utils.sleep(5000); // To ensure the providers are registered.
+			console.log(`Start schema compare tests`);
+
+		});
+		test('Schema compare dacpac comparision and script generation', async function () {
+			let service = await azdata.dataprotocol.getProvider<azdata.SchemaCompareServicesProvider>('MSSQL', azdata.DataProviderType.SchemaCompareServicesProvider);
+			assert(service, 'Schema Compare Service Provider is not available');
+
+			let source: azdata.SchemaCompareEndpointInfo = {
+				endpointType: azdata.SchemaCompareEndpointType.dacpac,
+				packageFilePath: path.join(__dirname, 'testData/Database1.dacpac'),
+				databaseName: 'database1',
+				ownerUri: '',
+			};
+			let target: azdata.SchemaCompareEndpointInfo = {
+				endpointType: azdata.SchemaCompareEndpointType.dacpac,
+				packageFilePath: path.join(__dirname, 'testData/Database2.dacpac'),
+				databaseName: 'database2',
+				ownerUri: '',
+			};
+
+			let schemaCompareResult = await service.schemaCompare(source, target, azdata.TaskExecutionMode.execute);
+			assertSchemaCompareResult(schemaCompareResult);
+
+			let status = await service.schemaCompareGenerateScript(schemaCompareResult.operationId, 'testDb', path.join(__dirname, 'testScript_dacpac.sql'), azdata.TaskExecutionMode.execute);
```

Comment:
Generate script shouldn't be an allowed operation if the target is a dacpac

Reply:
That's in UX. service doesn't care - right?

Reply:
The schema compare public API only supports generating the script when the target is a database. It looks like the 'testDb' being passed here is just a dummy database name?

Reply:
Yes